v2/brigadier/src/jobs.ts

```diff
@@ -20,33 +61,177 @@ export class Job {
+   * (the default) and Always.
+   */
   public imagePullPolicy = "IfNotPresent"
+  /** The working directory for the process running in the container. */
   public workingDirectory = ""
+  /**
+   * The command to run in the container. If not specified, the default entry point
+   * of the container is called.
+   * 
+   * Only the first element of the array is the actual command. Subsequent elements
+   * are treated as arguments. For example, a command of ["echo", "hello"] is
+   * equivalent to running 'echo hello'. A common convention is to use the command
+   * array for subcommands and the arguments array for argument values.
+   * 
+   * @example
+   * job.command = ["helm", "install"]
+   * job.arguments = ["stable/nginx", "-g"]
+   */
   public command: string[] = []
+  /**
+   * The arguments to pass to Container#command.  If the command includes arguments
+   * already, the arguments property are appended. A common convention is to use the command
+   * array for subcommands and the arguments array for argument values.
+   * 
+   * @example
+   * job.command = ["helm", "install"]
+   * job.arguments = ["stable/nginx", "-g"]
+   * 
+   */
   public arguments: string[] = []
+  /** Environment variables to set in the container. */
   public environment: { [key: string]: string } = {}
+  /**
+   * The path in the container's file system where, if applicable, the
+   * Brigade worker's shared workspace should be mounted. If empty (the default),
+   * the Job does not have access to the shared workspace.
+   */
   public workspaceMountPath = ""
+  /**
+   * The path in the container's file system where, if applicable,
+   * source code retrieved from a version control system repository should be
+	 * mounted. If empty (the default), Brigade will not mount any source
+   * code automatically.
+   */
   public sourceMountPath = ""
+  /**
+   * Whether the container should run with privileged permissions. This is
+   * typically required only for "Docker in Docker" scenarios where the
+   * container must run its own Docker daemon.
+   * 
+   * Privileged execution may be disallowed by Brigade project configuration.
+   * If so, the container will run unprivileged.
+   */
   public privileged = false
+  /**
+   * Whether the container should mount the host's Docker socket into its own
+   * file system. This is typically required only for "Docker-out-of-Docker" ("DooD")
+   * scenarios where the container needs to use the host's Docker daemon.
+   * This is strongly discouraged for almost all use cases.
+   * 
+   * Host Docker socket access may be disallowed by Brigade project configuration.
+   * If so, the container will run without such access.
+   */
   public useHostDockerSocket = false
 
+  /**
+   * Constructs a new Container.
+   * @param image The OCI reference to the container image
+   */
   constructor(image: string) {
     this.image = image
   }
 }
 
+/**
+ * The execution environment required by a Job.
```

Comment:
See related [comment](https://github.com/brigadecore/brigade/pull/1292/files?file-filters%5B%5D=.ts#r591634530)

Reply:
I feel comfortable with this level of abstraction, that's it's about the environments under which the job can run, but I can add some wording that makes it more specific to host selection if you want.